pkg/types/azure/validation/platform.go

```diff
@@ -68,18 +69,44 @@ func ValidatePlatform(p *azure.Platform, publish types.PublishingStrategy, fldPa
 	if p.DefaultMachinePlatform != nil {
 		allErrs = append(allErrs, ValidateMachinePool(p.DefaultMachinePlatform, "", p, nil, fldPath.Child("defaultMachinePlatform"))...)
 	}
-	if p.VirtualNetwork != "" {
-		if p.ComputeSubnet == "" {
-			allErrs = append(allErrs, field.Required(fldPath.Child("computeSubnet"), "must provide a compute subnet when a virtual network is specified"))
+	hasControlPlane := false
+	hasCompute := false
+	subnetSpecList := map[string]bool{}
+	for _, subnets := range p.Subnets {
+		if _, ok := subnetSpecList[subnets.Name]; ok {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("subnets"), subnets.Name, "duplicate value for subnet name"))
 		}
-		if p.ControlPlaneSubnet == "" {
-			allErrs = append(allErrs, field.Required(fldPath.Child("controlPlaneSubnet"), "must provide a control plane subnet when a virtual network is specified"))
+		subnetSpecList[subnets.Name] = true
+		switch subnets.Role {
+		case capz.SubnetControlPlane:
+			if hasControlPlane {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("subnets"), subnets.Name, "CAPZ currently does not support multiple control plane subnets"))
```

Comment:
```suggestion
				allErrs = append(allErrs, field.Invalid(fldPath.Child("subnets"), subnets.Name, "multiple control plane subnets are not currently supported"))
```

nit: capz is internal implementation so we should just say multiple control plane is not supported, right?